web/src/features/moreCast2/components/forecastHeader.test.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react'
+import { render, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import ForecastHeader from 'features/moreCast2/components/ForecastHeader'
+import { ColumnClickHandlerProps } from 'features/moreCast2/components/TabbedDataGrid'
+
+describe('ForecastHeader', () => {
+  const mockColumnClickHandlerProps: ColumnClickHandlerProps = {
+    colDef: { field: 'testForecast', headerName: 'test' },
+    contextMenu: null,
+    updateColumnWithModel: jest.fn(),
+    handleClose: jest.fn()
+  }
+
+  const renderForecastHeader = () => {
+    return render(
+      <ForecastHeader
+        colDef={{ field: 'testForecast', headerName: 'test' }}
+        columnClickHandlerProps={mockColumnClickHandlerProps}
+      />
+    )
+  }
+
+  const renderAndOpenMenu = async () => {
+    const rendered = renderForecastHeader()
+    const forecastButton = rendered.getByTestId('testForecast-column-header')
+    await userEvent.click(forecastButton)
+    await waitFor(() => expect(rendered.getByTestId('apply-to-column-menu')).toBeInTheDocument())
+    return rendered
+  }
+
+  it('should render an enabled forecast button without menu enabled by default', () => {
+    const { getByTestId, queryByTestId } = renderForecastHeader()
+
+    const forecastButton = getByTestId('testForecast-column-header')
+    expect(forecastButton).toBeInTheDocument()
+    expect(forecastButton).toBeEnabled()
+    expect(queryByTestId('apply-to-column-menu')).not.toBeInTheDocument()
+  })
+  it('should open apply to column menu when forecast button clicked', async () => {
```

Comment:
Is this test complete?

Reply:
Yep, it'll fail if the function it calls fails -- it's reused code for the next test

Reply:
Ah, I see now!